R/add_ppa_flag.R

```diff
@@ -0,0 +1,149 @@
+#' Add Potentially Preventable Admissions Marker
+#'
+#' @description This function takes a data frame input and determines, based on
+#' a combination of diagnostic codes and operation codes, whether an admission was
+#' preventable or not
+#' @param data A data frame
+#'
+#' @return A data frame to use as a lookup of PPAs
+#' @export
+add_ppa_flag <- function(data) {
+  matching_data <- data %>%
+    # Select out only the columns we need
+    dplyr::select(
+      .data$anon_chi, .data$cij_marker, .data$cij_pattype, .data$recid,
+      .data$op1a, .data$diag1, .data$diag2, .data$diag3, .data$diag4,
+      .data$diag5, .data$diag6
+    ) %>%
+    # Filter only recids and patient type where admission was preventable
+    dplyr::filter(.data$recid %in% c("01B", "02B", "04B", "GLS") & .data$cij_pattype == "Non-Elective") %>%
+    # We only want the first record in each cij, and we want to exclude empty cij and empty anon_chi
+    dplyr::group_by(.data$anon_chi, .data$cij_marker) %>%
+    dplyr::filter(dplyr::row_number() == 1 & !is.na(.data$cij_marker) & !is.na(.data$anon_chi)) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+
+      # Excluding operations are op1a codes from K01 to K50, K56, K60, and K61 (dental)
+      excluding_operation = stringr::str_sub(.data$op1a, 1, 3) %in%
+        purrr::reduce(list(
+          glue::glue("K{stringr::str_c(\"0\", c(1:9))}"),
+          glue::glue("K{as.character(c(10:50))}"),
```

Comment:
`paste0("K", 10:50)` or `glue::glue("K{10:50}")` are equivalent to `glue::glue("K{as.character(c(10:50))}")`